js/components/slider.js

```diff
@@ -0,0 +1,61 @@
+var React = require('react'),
+ReactDOM = require('react-dom'),
+Utils = require('./utils');
+
+var Slider = React.createClass({
+
+  componentDidMount: function() {
+    this.handleSliderColoring();
+  },
+
+  componentDidUpdate: function() {
+    this.handleSliderColoring();
```

Comment:
consider using `componentWillReceiveProps ` and check if the value changes, there are cases where `componentDidUpdate ` runs when there is no slider value change.

```
componentWillReceiveProps: function(nextProps) {
    if (nextProps.value != this.props.value) {
      this.handleSliderColoring();
    }
  },
```